pkg/model/firewall.go

```diff
@@ -42,72 +42,74 @@ type FirewallModelBuilder struct {
 var _ fi.ModelBuilder = &FirewallModelBuilder{}
 
 func (b *FirewallModelBuilder) Build(c *fi.ModelBuilderContext) error {
-	if err := b.buildNodeRules(c); err != nil {
+	nodeGroups, err := b.buildNodeRules(c)
+	if err != nil {
 		return err
 	}
-	if err := b.buildMasterRules(c); err != nil {
+
+	masterGroups, err := b.buildMasterRules(c, nodeGroups)
+	if err != nil {
 		return err
 	}
+
+	// We _should_ block per port... but:
+	// * It causes e2e tests to break
+	// * Users expect to be able to reach pods
+	// * If users are running an overlay, we punch a hole in it anyway
+	// re-using provided security group for the node
+	b.applyNodeToMasterBlockSpecificPorts(c, nodeGroups, masterGroups)
```

Comment:
Can you add the commented `//b.applyNodeToMasterAllowSpecificPorts(c)` back please?